lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm

```diff
@@ -117,6 +119,113 @@ sub list {
     $self->render(json => {JobTemplates => \@templates});
 }
 
+sub validate_yaml {
+    my ($self, $yaml, $validate_schema) = shift;
+
+    # Note: Using the schema filename; slurp'ed text isn't detected as YAML
+    my $schema_yaml = $self->app->home->child('assets', 'job.yaml')->to_string;
+    my $validator   = JSON::Validator->new;
+    my $schema;
+    my @errors;
+    if ($validate_schema) {
+        # Validate the schema: catches errors in type names and definitions
+        $validator = $validator->load_and_validate_schema($schema_yaml);
+        $schema    = $validator->schema;
+    }
+    else {
+        $schema = $validator->schema($schema_yaml);
+    }
+    if ($schema) {
+        # Note: Don't pass $schema here, that won't work
+        push @errors, $validator->validate(\%{$yaml});
+    }
+    else {
+        push @errors, "Failed to load schema";
+    }
+    @errors;
+}
+
+sub schedules {
+    my $self = shift;
+
+    my %yaml;
+    my $groups = $self->db->resultset("JobGroups")
+      ->search($self->param('id') ? {id => $self->param('id')} : undef, {select => [qw(id name)]});
+    while (my $group = $groups->next) {
+        my %group;
+        my $templates
+          = $self->db->resultset("JobTemplates")->search({group_id => $group->id}, {order_by => 'me.test_suite_id'});
+
+        # Always set the hash of test suites to account for empty groups
+        $group{architectures} = {};
+
+        my %machines;
+        my %prios;
+        # Extract products and tests per architecture
+        while (my $template = $templates->next) {
+            $group{products}{$template->product->name} = {
+                distri  => $template->product->distri,
+                flavor  => $template->product->flavor,
+                version => $template->product->version
+            };
+            my %test_suite;
+            $test_suite{machine} = $template->machine->name;
+            $machines{$template->product->arch}{$template->machine->name}++;
+            if ($template->prio) {
+                $test_suite{prio} = $template->prio;
+                $prios{$template->product->arch}{$template->prio}++;
```

Comment:
I would keep that way of dealing with the default prio for now. It makes sense to make the most frequent value the default.

However, like I've already mentioned in IRC, we actually already have a default which is deduced from the group or even parent group. But how we make that fit into the new approach we need to resolve at latest when implementing the import.

@coolo What do you think?

Reply:
IMO: just don't output what's not in the DB (aka is calculated from defaults). And as such: what is not in the input, don't store.

Reply:
So the biggest reason for having this heuristic is that values for `prio`, while technically there's a concept of a default, are defined explicitly more often than not. So what you get becomes a lot more verbose - since we want humans to copy, edit and paste these YAMLs I think it's worth the extra effort of making the export simpler as well. And the `machine` naturally has no default so it also benefits from this.

Reply:
The prio now effectively mirrors `JobGroups.default_priority` - prio will be omitted from the test suites in the YAML when it matches the default. As a side effect when importing the default priority can be updated based on the YAML as well. So now only the machine is a convenient heuristic.